src/solver/modeler/loadFiles/readLibraries.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include <antares/io/file.h>
+#include <antares/logs/logs.h>
+#include <antares/solver/modelConverter/modelConverter.h>
+#include <antares/solver/modelParser/parser.h>
+#include "antares/solver/modeler/loadFiles/loadFiles.h"
+
+namespace fs = std::filesystem;
+
+namespace Antares::Solver::LoadFiles
+{
+
+static Study::SystemModel::Library loadSingleLibrary(const fs::path& filePath)
+{
+    const std::string libraryStr = IO::readFile(filePath);
+
+    ModelParser::Parser parser;
+    ModelParser::Library libraryObj = parser.parse(libraryStr);
+    return ModelConverter::convert(libraryObj);
+}
+
+std::vector<Study::SystemModel::Library> loadLibraries(const fs::path& studyPath)
+{
+    std::vector<Study::SystemModel::Library> libraries;
+
+    const fs::path directoryPath = studyPath / "input" / "model-libraries";
+    for (const auto& entry: fs::directory_iterator(directoryPath))
+    {
+        if (entry.path().extension() != "yml")
+        {
+            logs.info() << "File ignored because of wrong extension: " << entry.path();
```

Comment:
OK but what is a valid extension ?
```suggestion
            logs.info() << entry.path() " ignored, only files having the `yml` extension are loaded";
```